connector-pagerduty/src/main/java/com/redhat/cloud/notifications/connector/pagerduty/PagerDutyCloudEventDataExtractor.java

```diff
@@ -30,22 +30,22 @@ public void extract(Exchange exchange, JsonObject cloudEventData) throws Illegal
 
         validatePayload(cloudEventData);
 
-        exchange.setProperty(TARGET_URL, cloudEventData.getString(URL));
+        String providedUrl = cloudEventData.getString(URL);
+        if (providedUrl == null || providedUrl.isEmpty()) {
+            exchange.setProperty(TARGET_URL, PAGERDUTY_EVENT_V2_URL);
+        } else {
+            exchange.setProperty(TARGET_URL, providedUrl);
+        }
```

Comment:
IIUC, this is needed to make the tests pass. I do not like it but I also consider this a low priority, so let's improve the tests and get rid of that code in a follow-up PR.

Reply:
come up with a solution, will include in future PR. removed an assertion from PagerDutyCloudEventDataExtractorTest that was unnecessary